src/components/Board.js

```diff
@@ -23,15 +36,15 @@ class Board extends Component {
       alert('Press Refresh')
     } else {
       if(this.state.board[index] === '') {
-        this.state.board[index] = this.state.currentTurn
+        let clonedBoard = this.state.board.slice(0)
```

Comment:
This is the state setting that you were doing improperly. I don't think it was causing issues, but it's good to avoid, since it might. 

All `this.state.board.slice(0)` does is clone the board array in the state. I can then manually change this clonedBoard array however I like, then run `this.setState({board: clonedBoard})`.

This is much better than manually changing the state without using `this.setState` like you were doing before.

Reply:
This is what Immutable.js probably helps you with – doing the cloning using some nice methods.